src/rewrite.rs

```diff
@@ -30,6 +31,23 @@ impl<T: Rewrite> Rewrite for ptr::P<T> {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) enum MacroErrorKind {
+    ParseFailure,
+    ReplaceMacroVariable,
+    Unknown,
+}
+
+impl fmt::Display for MacroErrorKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            MacroErrorKind::ParseFailure => write!(f, "(parse failure)"),
+            MacroErrorKind::ReplaceMacroVariable => write!(f, "(replacing macro variables with $)"),
+            MacroErrorKind::Unknown => write!(f, "(unknown reason)"),
```

Comment:
I replaced empty string with "(unknown reason)", and full error msg would look like 
```
Failed to format given macro(unknown reason) at: Span {....} 
```
I’d appreciate any other suggestions regarding the error message, though.

Reply:
I didn't realize that this `Display` impl would feed into `RewriteError::MacroFailure`s `Display` impl. I think that not specifying a reason would have been fine now that I think about it more.

Reply:
Then I'll just revert it to empty string :)